File: frontend/src/services/api/dailyLedger.ts
import api from '../api';

export interface DailyLedger {
  id: string;
  ledger_date: string;
  product_code: string;
  beginning_stock: number;
  total_inbound: number;
  total_outbound: number;
  adjustments: number;
  ending_stock: number;
  created_at: string;
  product?: {
    product_code: string;
    product_name: string;
    unit: string;
  };
}

export interface LedgerSummary {
  date: string;
  total_products: number;
  total_inbound: number;
  total_outbound: number;
  total_adjustments: number;
}

export const dailyLedgerAPI = {
  // 일일 수불부 조회
  getAll: (params?: { ledger_date?: string; product_code?: string }) => {
    return api.get<DailyLedger[]>('/daily-ledgers', { params });
  },

  // 일일 수불부 생성
  generate: (targetDate: string) => {
    return api.post<{ message: string; ledgers_created: number }>('/daily-ledgers/generate', null, {
      params: { target_date: targetDate }
    });
  },

  // 일일 수불부 요약 조회
  getSummary: (targetDate: string) => {
    return api.get<LedgerSummary>('/daily-ledgers/summary', {
      params: { target_date: targetDate }
    });
  }
};

File: backend/app/schemas/daily_ledger.py
"""
일일 수불부 스키마
"""
from typing import Optional
from datetime import date, datetime
from pydantic import BaseModel, Field


class DailyLedgerBase(BaseModel):
    """일일 수불부 기본 스키마"""
    ledger_date: date
    product_code: str
    beginning_stock: int
    total_inbound: int = 0
    total_outbound: int = 0
    adjustments: int = 0
    ending_stock: int


class DailyLedgerCreate(DailyLedgerBase):
    """일일 수불부 생성 스키마"""
    pass


class ProductInfo(BaseModel):
    """제품 정보"""
    product_code: str
    product_name: str
    unit: str


class DailyLedgerResponse(DailyLedgerBase):
    """일일 수불부 응답 스키마"""
    id: str
    created_at: str
    product: Optional[ProductInfo] = None
    
    class Config:
        from_attributes = True